src/cca/index.html

```diff
@@ -21,45 +25,108 @@
   <link href='generic_ui/style/global.css' rel='stylesheet' shim-shadowdom>
 
   <style>
-    paper-toolbar.uproxy {
-      background: #12a391;
-    }
-    .server-entry-card, .access-code-card {
-      background-color: white;
-      box-shadow: 1px 1px 3px #AAAAAA;
-      margin: 10px;
-      padding: 10px;
-    }
-    .access-code-card {
-      padding: 5px;
-    }
-    .access-code-card input {
-      width: 20em;
-    }
-    .page-content {
-      padding: 5px;
-    }
     @font-face {
       font-family: Quicksand;
       src: url(generic_ui/fonts/Quicksand-Regular.ttf);
     }
+    
     @font-face {
       font-family: Quicksand;
       src: url(generic_ui/fonts/Quicksand-Bold.ttf);
       font-weight: 700;
     }
-    html {
-      height: 100%;
-      position: relative;
-    }
+    
     body {
-      display: block;
-      margin: 0;
       font-family: Roboto, sans-serif;
-      font-size: 13px;
-      background-color: #eee;
+    }
+    
+    paper-toolbar.uproxy {
+      background: #12a391;
+    }
+
+    /*
+      Server cards use Flexbox to achieve the several
+      required instances of vertical centering.
+
+      It's a three column layout with the columns
+      comprising, from top to bottom:
+       - network name (toolbar)
+       - connect/disconnect button
+       - connection status
+
+      The top and bottom elements are configured with
+      an equal flex-grow attribute, leaving the button
+      sandwiched perfectly in the middle of the card.
+      
+      This interactive code snippet may help with
+      understanding the approach:
+        https://codepen.io/anon/pen/MbrpyK
+    */
+
+    #server-list {
       height: 100%;
     }
+    
+    #mainContainer::-webkit-scrollbar {
+      display: none;
+    }
+    
+    .server-card {
+      width: 100%;
+      display: flex;
+      flex-direction: column;
+    }
+    
+    .server-card .toolbar {
+      flex-grow: 1;
+      flex-basis: 0;
+    }
+    
+    .server-card .toolbar > div {
+      margin-top: 24px;
+      margin-left: 24px;
+    }
+    
+    .server-card .button {
+      align-self: center;
+    }
+    
+    .server-card .button > paper-icon-button {
+      width: 175px;
+      height: 175px;
+    }
+    
+    .server-card .status {
+      flex-grow: 1;
+      flex-basis: 0;
+      display: flex;
+      align-items: center;
+      justify-content: center;
+    }
+    
+    .server-card:not(:last-child) {
+      margin-bottom: 12px;
+    }
+  </style>
+
+  <style is="custom-style">
```

Comment:
I'm using it so I can re-use the Material Design colour `var`s.

I found some more info on this attribute here:
https://www.polymer-project.org/1.0/docs/api/custom-style